### ISSUE TITLE
Using each service's metric API to substitute System Management services

### ISSUE THREAD

**Issue by MonicaisHer** [edgexfoundry/edgex-ui-go#486]
UI is currently using[ SMA](https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/system-agent/2.0.0#/default/get_system_metrics) for tracking each service's metric. I suggest UI use [each service's metric API](https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-data/2.0.0#/default/get_metrics) instead. They have the same return value.

SMA is deprecated with the Ireland release. Although SMA is still in use, but it has been tagged for eventual replacement in the future release.

**Comment by badboy-huaqiao** [comment]
@MonicaisHer yes, it's a good advice, I have been thinking about this feature, but it needs more time to redesign  the GUI, so it's not finished before code freeze, keep  this issue open until it's done.

**Comment by bnevis-i** [comment]
@badboy-huaqiao What if as part of service registration the metrics API URL was added as additional service metadata?  Then, instead of hard-coding it into the UI it could just be queried from Consul?  

- Base URL -- http://localhost:port for snaps, http://service-name:port for docker, or https://kong-endpoint for secure services
- Service-prefix -- /core-data, for example
- Metrics sub-url -- /api/v2/metrics, for example, combine with the above.

If we did it right, security-proxy-setup could just walk consul and configure the forwards, which I think would simplify the code a bit.

**Comment by badboy-huaqiao** [comment]
@bnevis-i that's great if the registered services returned carried the metadata you mentioned above. 
but it might increase the workload on your team, as far as I know, services data returned by Consul's API will not contain such metadata.

Now the data structure of the registration service returned by the` go-mod-register `API looks like:
```
{
   "ServiceId":"core-metadata"
   "Host":"edgex-core-metadata"
   "Port": 59881
}
```
it's better to add some extra metadatas like `RoutePrefix` and `sub-url `， `RoutePrefix` is required which wil  be used both in secure and  insecure mode.

**Comment by bnevis-i** [comment]
I would file a request in go-mod-registry.  There is a Meta property https://www.consul.io/api-docs/agent/service#meta where we could add the additional information.

**Comment by badboy-huaqiao** [comment]
@bnevis-i that's great, looking forward to the updates. 
so now what about Kamakura? should we leave it unchanged or hide the system agent module on the GUI? does the SMA still exist in the Kamakura release?

**Comment by bnevis-i** [comment]
I don't know.  cc @jpwhitemn @lenny-intel   Want to weigh in?

**Comment by jpwhitemn** [comment]
I will be adding this issue to the architectural topics for the Levski planning meeting for May 16-19.  In the near term, the SMA service is not going away.  While deprecated, it can't go away until at least EdgeX 3.0.  So that gives us tome time to plan what is the path forward and what/how do we want to surface SMA info to tools like the GUI.

For now, GUI should take no action and we should add no additional features to any module or service.  We should address this in a consistent way once the community picks a path forward.

**Comment by badboy-huaqiao** [comment]
ok, remove the `Kamakura` tag but keep this issue open until the new features mentioned aboved is done.
